### DISCUSSION TITLE
How to set the home folder?

### DISCUSSION THREAD

**Discussion by buhtz** [jmcgeheeiv/pyfakefs#808] [Q&A]
Can I set the "home" folder `$HOME` on the faked filesystem?

Currently `pathlib.Path('~').resolve()` does result in '/~`.

**Comment by mrbean-bremen** [comment]
Hm, I'm not quite sure about the correct behavior. Testing this under Windows and under WSL actually gives the cwd + `"~"`, while `Path.home()` correctly returns the home directory. 
_EDIT:_ I checked this also under native Linux, and calling `pathlib.Path('~').resolve()` has the same result: cwd + "~". So, setting `$HOME` would not change anything both in the real and the fake filesystem for that call.

**Comment by mrbean-bremen** [comment]
To clarify: as far as I can see, the fake filesystem in this case behaves exactly like the real filesystem: `pathlib.Path('~').resolve()` returns the current working directory with the tilde as the last path parameter. Changing `$HOME` does not change this outcome.
If you want to access the home directory via `pathlib.Path`, you have to use `Path.home()` instead.

**Comment by mrbean-bremen** [comment]
As for the original question, how to set the home directory: setting `HOME` (or `USERPROFILE` under Windows) shall indeed change the home directory, as given by `pathlib.Path.home()`, or by  `pathlib.Path("~").expanduser()`.
